### PR TITLE
Update feature-gates.md to document renaming of feature gates

### PR DESCRIPTION
**Pull request by Porges**
A bunch of features were removed/renamed by https://github.com/kubernetes/kubernetes/pull/98243 and https://github.com/kubernetes/kubernetes/pull/101735, so document this.

### CODE REVIEW

File: content/en/docs/reference/command-line-tools-reference/feature-gates.md [hunk 1]
```diff
@@ -69,21 +69,21 @@ different Kubernetes components.
 | `CSIMigration` | `true` | Beta | 1.17 | |
 | `CSIMigrationAWS` | `false` | Alpha | 1.14 | |
 | `CSIMigrationAWS` | `false` | Beta | 1.17 | |
-| `CSIMigrationAWSComplete` | `false` | Alpha | 1.17 | |
+| `CSIMigrationAWSComplete` | `false` | Alpha | 1.17 | 1.20 |
```
Comment (tengqm) [inline content/en/docs/reference/command-line-tools-reference/feature-gates.md:72]: Where are these `CSIMigration***` gates going? Are they deprecated or graduated?
We should move them to the second table if so.
Another question: shouldn't the second version number be "1.21"?

**Comment by tengqm** [comment]
Looks like this one should target `dev-1.22` branch rather than `master`?

File: content/en/docs/reference/command-line-tools-reference/feature-gates.md (hunk 1 above)
Comment (Porges) [reply content/en/docs/reference/command-line-tools-reference/feature-gates.md:72]: They were “renamed” so they have been removed entirely. I think the last column is meant to be the last version they are available? They are all gone in 1.21.

**Comment by Porges** [comment]
> Looks like this one should target `dev-1.22` branch rather than `master`?

They’re not available in 1.21, so the docs are already out of sync with what is released.

File: content/en/docs/reference/command-line-tools-reference/feature-gates.md (hunk 1 above)
Comment (Porges) [reply content/en/docs/reference/command-line-tools-reference/feature-gates.md:72]: Up above the last column is "the last Kubernetes release in which you can still use a feature gate"

Comment (tengqm) [reply content/en/docs/reference/command-line-tools-reference/feature-gates.md:72]: Take `CSIMigrationAWSComplete` as an example, if they were effective in 1.20, but "deprecated" in 1.21. We should remove the lines from the first table and add the following lines to the second table.  The first table only maintains the gates that are still effective, aka, `alpha` or `beta`.

```suggestion
| `CSIMigrationAWSComplete` | `false` | Alpha | 1.17 | 1.20 |
| `CSIMigrationAWSComplete` | `false` | Deprecated | 1.21 | - |
```

Comment (Porges) [reply content/en/docs/reference/command-line-tools-reference/feature-gates.md:72]: @tengqm in this case they are not simply deprecated they are removed entirely. Should I move them to the second table with no "Deprecated" line?

**Comment by tengqm** [comment]
Hopefully, we will avoid renaming gates in the future. It causes more confusion than clarification.
Given that this has already happened, if we are reluctant to say the "xyzComplete" gate has been deprecated, we can explicitly say "renamed to BlahBlah". We never know which version our users are using so we'd better make it crystal clear.

**Comment by Porges** [comment]
@tengqm, good idea. I have added "renamed" entries into the table.

File: content/en/docs/reference/command-line-tools-reference/feature-gates.md [hunk 2]
```diff
@@ -671,6 +652,38 @@ Each feature gate is designed for enabling/disabling a specific feature:
   to `IngressClass.spec.parameters`.
 - `Initializers`: Allow asynchronous coordination of object creation using the
   Initializers admission plugin.
+- `InTreePluginAWSUnregister`: Stops registering the EBS in-tree plugin in
```
Comment (Jiawei0227) [inline content/en/docs/reference/command-line-tools-reference/feature-gates.md:655]: The InTreePluginXXXUnregister is not only a rename. This is to decouple unregistering a plugin from CSI migration. Previously if you want to disable an intree plugin. There is no flag that can do it. The only way is to enable CSI migration and enable this Complete flag.

Now with the new flag. You can set it directly to disable an intree plugin without enabling CSI migration.

See https://github.com/kubernetes/website/pull/28223 for more context.

**Comment by Jiawei0227** [comment]
I guess there is a confusing of the feature gate here. This is not simply a rename. The function is actually different that is why we have the new name. For the original PR: https://github.com/kubernetes/kubernetes/pull/98243

I also have another website update PR here https://github.com/kubernetes/website/pull/28223. I guess it is a duplicate of this one.

File: content/en/docs/reference/command-line-tools-reference/feature-gates.md [hunk 3]
```diff
@@ -128,6 +122,12 @@ different Kubernetes components.
 | `HugePageStorageMediumSize` | `true` | Beta | 1.19 | |
 | `IndexedJob` | `false` | Alpha | 1.21 | |
 | `IngressClassNamespacedParams` | `false` | Alpha | 1.21 | |
+| `InTreePluginAWSUnregister` | `false` | Alpha | 1.21 | |
+| `InTreePluginAzureDiskUnregister` | `false` | Alpha | 1.21 | |
+| `InTreePluginAzureFileUnregister` | `false` | Alpha | 1.21 | |
+| `InTreePluginGCEUnregister` | `false` | Alpha | 1.21 | |
+| `InTreePluginOpenStackUnregister` | `false` | Alpha | 1.21 | |
+| `InTreePluginvSphereUnregister` | `false` | Beta | 1.21 | |
```
Comment (Jiawei0227) [inline content/en/docs/reference/command-line-tools-reference/feature-gates.md:130]: This is alpha

File: content/en/docs/reference/command-line-tools-reference/feature-gates.md [hunk 4]
```diff
@@ -225,6 +225,19 @@ different Kubernetes components.
 | `CSIPersistentVolume` | `false` | Alpha | 1.9 | 1.9 |
 | `CSIPersistentVolume` | `true` | Beta | 1.10 | 1.12 |
 | `CSIPersistentVolume` | `true` | GA | 1.13 | - |
+| `CSIMigrationAWSComplete` | `false` | Alpha | 1.17 | 1.20 |
+| `CSIMigrationAWSComplete` | `false` | Renamed to `InTreePluginAWSUnregister` | 1.21 | — |
```
Comment (Jiawei0227) [inline content/en/docs/reference/command-line-tools-reference/feature-gates.md:229]: This is not only a rename. See my comments above.

ditto

**Review by Jiawei0227 — COMMENTED** [review]
I guess if we are in favor of this PR, can you help to change the meaning of the feature gates? Thanks!

**Comment by tengqm** [comment]
/approve cancel

Looks like we don't have a consensus whether this change is only about a rename.

**Comment by jimangel** [comment]
I lack the technical review but if this merges it could impact the `dev-1.22` branch.

/cc @PI-Victor

**Comment by Porges** [comment]
Close in favour of #28223

**Pull request closed without merging**